rust/rpmostree-client/src/lib.rs

```diff
@@ -97,6 +97,22 @@ impl Deployment {
             Err(format!("No {} metadata key", k).into())
         }
     }
+
+    pub fn get_base_manifest(&self) -> Result<Option<oci_spec::image::ImageManifest>> {
+        if self.container_image_reference.is_none() {
+            return Ok(None);
+        }
+
+        let manifest = self.base_commit_meta.get("ostree.manifest");
+        if let Some(inner) = manifest {
+            let deser = String::deserialize(inner)?;
```

Comment:
I think it'd be cleaner to do something like:

```
if let Some(serde_json::Value::String(inner))) {
            let manifest: oci_spec::image::ImageManifest = serde_json::from_str(&inner)?;
            return Ok(Some(manifest));
}
```

or so, this `String::deserialize` is duplicating the value unnecessarily.